articles/customization-using-bulma/index.md

```diff
@@ -0,0 +1,237 @@
+/>
+```
+
+- Download the npm package from the npm repository.
+
+```
+npm install bulma
+```
+
+- We can also download the `bulma.css` file from [here](https://github.com/jgthms/bulma/tree/master/css) and include it in our webpage using the `<link>` tag.
+
+```html
+<link rel="stylesheet" href="bulma.css" />
+```
+
+### Column System
+
+One of the best features of Bulma is the responsive column grid system. For building a columns layout,
+
+- Every row in Bulma is defined by a `columns` container.
+- Every column should be defined as a `column` and should be nested within the `columns` container.
+
+For example,
+
+```html
+<div class="columns">
+  <div class="column">1st column</div>
+  <div class="column">2nd column</div>
+  <div class="column">3rd column</div>
+</div>
+```
+
+This snippet would create a responsive column layout with 3 columns of equal width. You can also specify the column size by using the size classes.
+
+```html
+<div class="columns">
+  <div class="column is-half">1st column</div>
+  <div class="column is-one-quarter">2nd column</div>
+  <div class="column">3rd column</div>
+</div>
+```
+
+The above snippet would create three columns. One column occupies half the space, and the other two occupy a quarter of the space. For more options on columns, visit [this link](https://bulma.io/documentation/columns/).
+
+### Components
+
+Bulma comes with many in-built UI components. Let's try to create a card component.
+
+```html
+<div class="card">
+  <div class="card-content">
+    <p class="title">
+      Lorem ipsum dolor, sit amet consectetur adipisicing elit. Architecto,
+      corrupti.
+    </p>
+    <p class="subtitle">Lorem Ipsum.</p>
+  </div>
+  <footer class="card-footer">
+    <p class="card-footer-item">
+      <span>Footer-Left</span>
+    </p>
+    <p class="card-footer-item">
+      <span>Footer-Right</span>
+    </p>
+  </footer>
+</div>
+```
+
+This code snippet would create a card component that would look like this:
+
+![Bulma Card](/engineering-education/customization-using-bulma/card.png)
+
+Bulma has many other components, for which the documentation can be viewed [here](https://bulma.io/documentation/components/).
+
+### Customizing Variables
+
+For customizing Bulma, there are four kinds or levels of variables:
+
+- Initial Variables -- There are global variables with literal values.
+- Derived Variables -- These are derived from other variables.
+- Generic Variables -- HTML elements that carry no CSS class.
+- Element Variables -- variables that are specific to a bulma element.
+
+Now, for customizing these Bulma Variables, let us set up sass and bulma. Sass is a CSS preprocessor that adds extra features like variables and loops. It is an extension to CSS that improves the stylesheet language and reduces repetition.
+
+- Initiate a node project.
+
+```
+npm init
+```
+
+- Install bulma and sass.
+
+```
+npm i bulma sass
+```
+
+- Now, to get started, create three files.
```

Comment:
create the three files below would be more semantic.

Reply:
Done, change made!